### ISSUE TITLE
jsk_2016_01_baxter_apcのrosdep installでエラーが出る

### ISSUE THREAD

**Issue by ban-masa** [start-jsk/jsk_apc#1244]
jsk_2016_01_baxter_apcのrosdep install中に以下のエラーが出ました。Cythonのバージョンについてエラーが出ているようです。
```
$ rosdep install -y -r --from-paths . --ignore-src
WARNING: Package "ompl" does not follow the version conventions. It should not contain leading zeros (unless the number is 0).
executing command [sudo -H pip install -U scikit-image]
Downloading/unpacking scikit-image
  Downloading scikit-image-0.12.3.tar.gz (20.7MB): 20.7MB downloaded
  Running setup.py (path:/tmp/pip_build_root/scikit-image/setup.py) egg_info for package scikit-image
    Unable to find pgen, not compiling formal grammar.
    No eggs found in /tmp/easy_install-wJfz6j/Cython-0.24/egg-dist-tmp-DO5lg8 (setup script problem?)
    Traceback (most recent call last):
      File "<string>", line 17, in <module>
      File "/tmp/pip_build_root/scikit-image/setup.py", line 146, in <module>
        **extra
      File "/usr/lib/python2.7/dist-packages/numpy/distutils/core.py", line 126, in setup
        dist = setup(**new_attr)
      File "/usr/lib/python2.7/dist-packages/numpy/distutils/core.py", line 169, in setup
        return old_setup(**new_attr)
      File "/usr/lib/python2.7/distutils/core.py", line 111, in setup
        _setup_distribution = dist = klass(attrs)
      File "/usr/lib/python2.7/dist-packages/numpy/distutils/numpy_distribution.py", line 16, in __init__
        Distribution.__init__(self, attrs)
      File "/usr/lib/python2.7/dist-packages/setuptools/dist.py", line 239, in __init__
        self.fetch_build_eggs(attrs.pop('setup_requires'))
      File "/usr/lib/python2.7/dist-packages/setuptools/dist.py", line 264, in fetch_build_eggs
        replace_conflicting=True
      File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 628, in resolve
        raise DistributionNotFound(req)
    pkg_resources.DistributionNotFound: cython>=0.21
    Complete output from command python setup.py egg_info:
    Unable to find pgen, not compiling formal grammar.

No eggs found in /tmp/easy_install-wJfz6j/Cython-0.24/egg-dist-tmp-DO5lg8 (setup script problem?)

Traceback (most recent call last):

  File "<string>", line 17, in <module>

  File "/tmp/pip_build_root/scikit-image/setup.py", line 146, in <module>

    **extra
  File "/usr/lib/python2.7/dist-packages/numpy/distutils/core.py", line 126, in setup

    dist = setup(**new_attr)

  File "/usr/lib/python2.7/dist-packages/numpy/distutils/core.py", line 169, in setup

    return old_setup(**new_attr)

  File "/usr/lib/python2.7/distutils/core.py", line 111, in setup

    _setup_distribution = dist = klass(attrs)

  File "/usr/lib/python2.7/dist-packages/numpy/distutils/numpy_distribution.py", line 16, in __init__

    Distribution.__init__(self, attrs)

  File "/usr/lib/python2.7/dist-packages/setuptools/dist.py", line 239, in __init__

    self.fetch_build_eggs(attrs.pop('setup_requires'))

  File "/usr/lib/python2.7/dist-packages/setuptools/dist.py", line 264, in fetch_build_eggs

    replace_conflicting=True

  File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 628, in resolve

    raise DistributionNotFound(req)

pkg_resources.DistributionNotFound: cython>=0.21

----------------------------------------
Cleaning up...
Command python setup.py egg_info failed with error code 1 in /tmp/pip_build_root/scikit-image
Storing debug log for failure in /root/.pip/pip.log
ERROR: the following rosdeps failed to install
  pip: command [sudo -H pip install -U scikit-image] failed
  pip: Failed to detect successful installation of [scikit-image]
```

**Comment by ban-masa** [comment]
```sudo pip install cython```でcythonを入れるとversion2.4のcythonを入れると```rosdep install -y -r --from-paths . --ignore-src```がうまくいきました

**Comment by wkentaro** [comment]
@ban-masa Please update https://github.com/start-jsk/jsk_apc/blob/master/doc/tips_faq.rst with reference to this issue.

**Comment by k-okada** [comment]
@wkentaro is there way to install cython from pip in rosdep ? or since we did not have this error before, so we need to specify the version of scikit-image? or fix scikit-image to install latest cython?

**Comment by wkentaro** [comment]
Actually, I don't have error on **sheeta** PC.
There is no cython, but `scikit-image==0.12.3` is installed.
```
% pip freeze | grep -i cython | wc -l
0
```
```
% pip freeze | grep -i scikit-image
scikit-image==0.12.3
```
And also, cython is not required when installing via pip: https://github.com/scikit-image/scikit-image/blob/master/setup.py#L135

So, now I think the issue is caused by the old setuptools on @ban-masa PC. (version 20.6.7 on sheeta PC)
@ban-masa could you please try below to test this?
```
$ sudo pip uninstall cython
$ sudo pip uninstall scikit-image

$ roscd jsk_apc/..
$ rosdep install --from-paths . -y -r -i  # it should fail

$ sudo pip install -U pip setuptools
$ pip list | grep pip
pip (8.1.1)
$ pip list | grep setuptools
setuptools (20.6.7)

$ roscd jsk_apc/..
$ rosdep install --from-paths . -y -r -i  # it should success
```

**Comment by ban-masa** [comment]
I did those commands.
First rosdep install failed due to the same error.
And the result is below.
```
$ sudo pip install -U pip setuptools
Downloading/unpacking pip from https://pypi.python.org/packages/py2.py3/p/pip/pip-8.1.1-py2.py3-none-any.whl#md5=22db7b6a517a09c29d54a76650f170eb
  Downloading pip-8.1.1-py2.py3-none-any.whl (1.2MB): 1.2MB downloaded
Downloading/unpacking setuptools from https://pypi.python.org/packages/py2.py3/s/setuptools/setuptools-20.7.0-py2.py3-none-any.whl#md5=9e0248519bf3e19ca6b3c217dc633ba8
  Downloading setuptools-20.7.0-py2.py3-none-any.whl (508kB): 508kB downloaded
Installing collected packages: pip, setuptools
  Found existing installation: pip 1.5.4
    Not uninstalling pip at /usr/lib/python2.7/dist-packages, owned by OS
  Found existing installation: setuptools 3.3
    Not uninstalling setuptools at /usr/lib/python2.7/dist-packages, owned by OS
Successfully installed pip setuptools
Cleaning up...
[~/ros_ws/ws_jsk_apc/src/start-jsk/jsk_apc]
$ pip list | grep pip
pip (1.5.4)
  File "/usr/lib/python2.7/dist-packages/pip/basecommand.py", line 122, in main
  File "/usr/lib/python2.7/dist-packages/pip/commands/list.py", line 80, in run
  File "/usr/lib/python2.7/dist-packages/pip/commands/list.py", line 142, in run_listing
  File "/usr/lib/python2.7/dist-packages/pip/commands/list.py", line 151, in output_package_listing
  File "/usr/lib/python2.7/dist-packages/pip/util.py", line 366, in dist_is_editable
  File "/usr/lib/python2.7/dist-packages/pip/__init__.py", line 286, in from_dist
Storing debug log for failure in /home/ban-masa/.pip/pip.log
[~/ros_ws/ws_jsk_apc/src/start-jsk/jsk_apc]
$ pip list | grep setuptools
[~/ros_ws/ws_jsk_apc/src/start-jsk/jsk_apc]
$ pip list | grep setuptools
[~/ros_ws/ws_jsk_apc/src/start-jsk/jsk_apc]
$ roscd jsk_apc/..
README.md  archives  doc  images  jsk_2015_05_baxter_apc  jsk_2016_01_baxter_apc  jsk_apc  jsk_apc2015_common  jsk_apc2016_common  kinect2.rosinstall
[~/ros_ws/ws_jsk_apc/src/start-jsk/jsk_apc]
$ rosdep install --from-paths . -y -r -i
WARNING: Package "ompl" does not follow the version conventions. It should not contain leading zeros (unless the number is 0).
executing command [sudo -H pip install -U scikit-image]
Collecting scikit-image
/usr/local/lib/python2.7/dist-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:315: SNIMissingWarning: An HTTPS request has been made, but the SNI (Subject Name Indication) extension to TLS is not available on this platfo
rm. This may cause the server to present an incorrect TLS certificate, which can cause validation failures. For more information, see https://urllib3.readthedocs.org/en/latest/security.html#snimissingwarning.
  SNIMissingWarning
/usr/local/lib/python2.7/dist-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:120: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may caus
e certain SSL connections to fail. For more information, see https://urllib3.readthedocs.org/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
  Downloading scikit_image-0.12.3-cp27-cp27mu-manylinux1_x86_64.whl (28.0MB)
    100% |████████████████████████████████| 28.0MB 61kB/s 
Requirement already up-to-date: networkx>=1.8 in /usr/local/lib/python2.7/dist-packages (from scikit-image)
Requirement already up-to-date: dask[array]>=0.5.0 in /usr/local/lib/python2.7/dist-packages (from scikit-image)
Requirement already up-to-date: six>=1.7.3 in /usr/local/lib/python2.7/dist-packages (from scikit-image)
Requirement already up-to-date: pillow>=2.1.0 in /usr/local/lib/python2.7/dist-packages (from scikit-image)
Requirement already up-to-date: decorator>=3.4.0 in /usr/local/lib/python2.7/dist-packages (from networkx>=1.8->scikit-image)
Requirement already up-to-date: numpy in /usr/local/lib/python2.7/dist-packages (from dask[array]>=0.5.0->scikit-image)
Requirement already up-to-date: toolz>=0.7.2 in /usr/local/lib/python2.7/dist-packages (from dask[array]>=0.5.0->scikit-image)
Installing collected packages: scikit-image
Successfully installed scikit-image-0.12.3
/usr/local/lib/python2.7/dist-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:120: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may caus
e certain SSL connections to fail. For more information, see https://urllib3.readthedocs.org/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
#All required rosdeps installed successfully
```

**Comment by wkentaro** [comment]
Do you know why it successfully downloads scikit-image?
It seems that pip and setuptools is not upgraded in your log.
```
Installing collected packages: pip, setuptools
  Found existing installation: pip 1.5.4
    Not uninstalling pip at /usr/lib/python2.7/dist-packages, owned by OS
  Found existing installation: setuptools 3.3
    Not uninstalling setuptools at /usr/lib/python2.7/dist-packages, owned by OS
Successfully installed pip setuptools
```

**Comment by ban-masa** [comment]
It seems that setuptools has installed in another directory too.
```
[~/ros_ws/ws_jsk_apc/src/start-jsk/jsk_apc]
$ pip show setuptools
---
Name: setuptools
Version: 20.7.0
Location: /usr/local/lib/python2.7/dist-packages
Requires: 
```

**Comment by wkentaro** [comment]
I see, so setuptools has been upgraded.
So the reason of the error was the outdated setuptools.

**Comment by wkentaro** [comment]
I will update the doc.

**Comment by wkentaro** [comment]
Closed via #1257
